File: python/Production/solver/RotationTree.py
class RotationTree:

       def __init__():
            self.c_array = list()

        def addRotationLinkedList(self, p_list):
            #Console.Write("FFFFF")
            self.c_array.append(p_list.getCopy())

        def getSize(self):
            return len(self.c_array)

        def getRotationSequence(self, p_index):
            return self.c_array[p_index]

        @staticmethod
        def getRotationTreeFromFile(p_File):
           myTree = RotationTree()
           l_rotationLinkedList = RotationSequence()
           while l_rotationLinkedList.readFromFile(p_File):
                myTree.addRotationLinkedList(l_rotationLinkedList)
            return myTree

File: csharp/Production/solver/RotationTree.cs
using cube;
using System.Collections.Generic;


namespace solver
{

    public class RotationTree
    {
        List<RotationSequence> c_array = new List<RotationSequence>();
        public RotationTree()
        {
        }

        public void addRotationLinkedList(RotationSequence p_list)
        {
            //Console.Write("FFFFF");
            c_array.Add(p_list.getCopy());
        }

        public int getSize()
        {
            return c_array.Count;
        }

        public RotationSequence getRotationSequence(int p_index)
        {
            return c_array[p_index];
        }

        public static RotationTree getRotationTreeFromFile(RubikFileReader p_File)
        {
            RotationTree myTree = new RotationTree();
            RotationSequence l_rotationLinkedList = new RotationSequence();
            while (l_rotationLinkedList.readFromFile(p_File))
            {
                myTree.addRotationLinkedList(l_rotationLinkedList);
            }
            return myTree;

        }
    }
}
